subsys/net/ip/tcp2.c

```diff
@@ -302,13 +302,18 @@ static void tcp_send(struct net_pkt *pkt)
 	if (is_6lo_technology(pkt)) {
 		struct net_pkt *new_pkt;
 
-		new_pkt = net_pkt_clone(pkt, TCP_PKT_ALLOC_TIMEOUT);
+		new_pkt = tcp_pkt_clone(pkt);
 		if (!new_pkt) {
+			/* The caller of this func assumes that the net_pkt
+			 * is consumed here, so make sure that the unref in
+			 * out: will free the net_pkt.
```

Comment:
> make sure that the unref in
			 * out: will free the net_pkt.

Apparently, there're typos in this comment, which make it hard to understand it. Was it meant something like: "make sure that we unref it now, that will free the net_pkt" ?

Reply:
Actually there was no typos here. The `out:` refers to the label at the end of the func. I will reword the comment so it is clear.